include/cxx.h

```diff
@@ -383,91 +476,68 @@ class Option<Box<T>> {
 
   // Internal API only intended for the cxxbridge code generator.
-  Vec(unsafe_bitcopy_t, const Vec &) noexcept;
-
+  Option(unsafe_bitcopy_t, const Option &) noexcept;
 private:
-  void reserve_total(std::size_t new_cap) noexcept;
-  void set_len(std::size_t len) noexcept;
-  void drop() noexcept;
-
-  friend void swap(Vec &lhs, Vec &rhs) noexcept { lhs.swap(rhs); }
-
-  // Size and alignment statically verified by rust_vec.rs.
   std::array<std::uintptr_t, 3> repr;
-};
-#endif // CXXBRIDGE1_RUST_VEC
-
-#ifndef CXXBRIDGE1_RUST_FN
-// https://cxx.rs/binding/fn.html
-template <typename Signature>
-class Fn;
 
-template <typename Ret, typename... Args>
-class Fn<Ret(Args...)> final {
-public:
-  Ret operator()(Args... args) const noexcept;
-  Fn operator*() const noexcept;
+  void drop() noexcept;
+};
 
-private:
-  Ret (*trampoline)(Args..., void *fn) noexcept;
-  void *fn;
+#define OPTION_DIRECT_DECL(INNER, REPR, COUNT)                    \
+template<>                                                        \
+class Option<INNER> {                                             \
+public:                                                           \
+  Option() noexcept;                                              \
+  Option(Option&&) noexcept;                                      \
+  Option(INNER&&) noexcept;                                       \
+  ~Option() noexcept;                                             \
+                                                                  \
+  Option<INNER>& operator=(Option&&) noexcept;                    \
+  const INNER& operator*() const noexcept;                        \
+  const INNER* operator->() const noexcept;                       \
+  INNER& operator*() noexcept;                                    \
+  INNER* operator->() noexcept;                                   \
+                                                                  \
+  bool operator==(const Option &) const noexcept;                 \
+  bool operator!=(const Option &) const noexcept;                 \
+                                                                  \
+  bool has_value() const noexcept;                                \
+  const INNER& value() const noexcept;                            \
+  INNER& value() noexcept;                                        \
+  void reset();                                                   \
+  void set(INNER) noexcept;                                       \
+                                                                  \
+  Option(unsafe_bitcopy_t, const Option &) noexcept;              \
+private:                                                          \
+  std::array<REPR, COUNT> repr;                                                      \
+                                                                  \
+  void drop() noexcept;                                           \
 };
-#endif // CXXBRIDGE1_RUST_FN
+
+OPTION_DIRECT_DECL(String, std::uintptr_t, 3)
```

Comment:
Looks a little silly to use a macro for just 1 impl, but this will be reused in a follow up when I support Option<primitive>